cartographer/mapping_2d/local_trajectory_builder.cc

```diff
@@ -223,64 +193,29 @@ LocalTrajectoryBuilder::pose_estimate() const {
 
 void LocalTrajectoryBuilder::AddImuData(const sensor::ImuData& imu_data) {
   CHECK(options_.use_imu_data()) << "An unexpected IMU packet was added.";
-
-  InitializeImuTracker(imu_data.time);
-  Predict(imu_data.time);
-  imu_tracker_->AddImuLinearAccelerationObservation(
-      imu_data.linear_acceleration);
-  imu_tracker_->AddImuAngularVelocityObservation(imu_data.angular_velocity);
+  InitializeExtrapolator(imu_data.time);
+  extrapolator_->AddImuData(imu_data);
 }
 
 void LocalTrajectoryBuilder::AddOdometerData(
     const common::Time time, const transform::Rigid3d& odometer_pose) {
-  if (imu_tracker_ == nullptr) {
-    // Until we've initialized the IMU tracker we do not want to call Predict().
-    LOG(INFO) << "ImuTracker not yet initialized.";
+  if (extrapolator_ == nullptr) {
+    // Until we've initialized the extrapolator we cannot add odometry data.
+    LOG(INFO) << "Extrapolator not yet initialized.";
     return;
   }
-
-  Predict(time);
-  if (!odometry_state_tracker_.empty()) {
-    const auto& previous_odometry_state = odometry_state_tracker_.newest();
-    const transform::Rigid3d delta =
-        previous_odometry_state.odometer_pose.inverse() * odometer_pose;
-    const transform::Rigid3d new_pose =
-        previous_odometry_state.state_pose * delta;
-    odometry_correction_ = pose_estimate_.inverse() * new_pose;
-  }
-  odometry_state_tracker_.AddOdometryState(
-      {time, odometer_pose, pose_estimate_ * odometry_correction_});
-}
-
-void LocalTrajectoryBuilder::InitializeImuTracker(const common::Time time) {
-  if (imu_tracker_ == nullptr) {
-    imu_tracker_ = common::make_unique<mapping::ImuTracker>(
-        options_.imu_gravity_time_constant(), time);
-  }
+  extrapolator_->AddOdometryData(sensor::OdometryData{time, odometer_pose});
 }
 
-void LocalTrajectoryBuilder::Predict(const common::Time time) {
-  CHECK(imu_tracker_ != nullptr);
-  CHECK_LE(time_, time);
-  const double last_yaw = transform::GetYaw(imu_tracker_->orientation());
-  imu_tracker_->Advance(time);
-  if (time_ > common::Time::min()) {
-    const double delta_t = common::ToSeconds(time - time_);
-    // Constant velocity model.
-    const Eigen::Vector3d translation =
-        pose_estimate_.translation() +
-        delta_t *
-            Eigen::Vector3d(velocity_estimate_.x(), velocity_estimate_.y(), 0.);
-    // Use the current IMU tracker roll and pitch for gravity alignment, and
-    // apply its change in yaw.
-    const Eigen::Quaterniond rotation =
-        Eigen::AngleAxisd(
-            transform::GetYaw(pose_estimate_.rotation()) - last_yaw,
-            Eigen::Vector3d::UnitZ()) *
-        imu_tracker_->orientation();
-    pose_estimate_ = transform::Rigid3d(translation, rotation);
+void LocalTrajectoryBuilder::InitializeExtrapolator(const common::Time time) {
+  if (extrapolator_ != nullptr) {
+    return;
   }
-  time_ = time;
+  constexpr double kExtrapolationEstimationTimeSec = 0.001;  // 1 ms
```

Comment:
A comment here about what this means would be useful. Based on my understanding, this should perhaps not even be a parameter but rather just a constant in the PoseExtrapolator?